controller/api/public/gateways.go

```diff
@@ -52,7 +55,34 @@ func buildGatewaysRequestLabels(req *pb.GatewaysRequest) (labels model.LabelSet,
 	return labels, groupBy
 }
 
-func processPrometheusResult(results []promResult) map[string]*pb.GatewaysTable_Row {
+
+// this function returns a map of gateways to the number of services using them
+func (s *grpcServer) getNumServicesMap() (map[string]uint64, error) {
+
+	results := make(map[string]uint64)
+	selector := 	fmt.Sprintf("%s,!%s", k8s.MirroredResourceLabel,  k8s.MirroredGatewayLabel)
+	services, err := s.k8sAPI.Client.CoreV1().Services(corev1.NamespaceAll).List(metav1.ListOptions{LabelSelector:selector})
+	if err != nil {
+		return nil, err
+	}
+
+	for _, svc := range services.Items {
+		if svc.Labels != nil {
```

Comment:
Given we're filtering on `k8s.MirroredResourceLabel`, wouldn't this always be non-nil?